src/expr/src/expr/expr_binary_nonnull.rs

```diff
@@ -307,56 +289,60 @@ pub fn new_binary_expr(
             gen_binary_expr_atm! {
                 gen_atm_impl,
                 l, r, ret,
+                general_add,
                 {
-                    General => general_add,
-                    (Timestamp, Interval) => timestamp_interval_add,
-                    (Interval, Timestamp) => interval_timestamp_add,
-                    (Interval, Date) => interval_date_add,
-                    (Date, Interval) => date_interval_add,
-                    (Interval, Interval) => general_add,
+                    {timestamp, interval, timestamp, timestamp_interval_add},
+                    {interval, timestamp, timestamp, interval_timestamp_add},
+                    {interval, date, timestamp, interval_date_add},
+                    {date, interval, timestamp,  date_interval_add},
+                    {interval, interval, interval,  general_add},
                 },
             }
         }
         Type::Subtract => {
             gen_binary_expr_atm! {
                 gen_atm_impl,
                 l, r, ret,
+                general_sub,
                 {
-                    General => general_sub,
-                    (Timestamp, Timestamp) => timestamp_timestamp_sub,
-                    (Timestamp, Interval) => timestamp_interval_sub,
-                    (Date, Date) => date_date_sub,
-                    (Date, Interval) => date_interval_sub,
-                    (Interval, Interval) => general_sub,
+                    {timestamp, timestamp, interval, timestamp_timestamp_sub},
+                    {timestamp, interval, timestamp, timestamp_interval_sub},
+                    {date, date, int32, date_date_sub},
+                    {date, interval, timestamp,  date_interval_sub},
+                    {interval, interval, interval,  general_sub},
                 },
             }
         }
         Type::Multiply => {
             gen_binary_expr_atm! {
                 gen_atm_impl,
                 l, r, ret,
+                general_mul,
                 {
-                    General => general_mul,
-                    (Interval, Int) => interval_int_mul,
-                    (Int, Interval) => int_interval_mul,
+                    {interval, int16, interval,  interval_int_mul},
```

Comment:
```suggestion
                    {interval, int16, interval, interval_int_mul},
```
extra space? 🤪 Also I'd prefer to have a space after brace `{`